src/components/track-tile/TrackTile.tsx

```diff
@@ -1,106 +1,215 @@
+  Pressable,
+  StyleSheet
+} from 'react-native'
+import { useDispatch, useSelector } from 'react-redux'
 
+import { BaseStackParamList } from 'app/components/app-navigator/types'
 import { TrackTileProps } from 'app/components/track-tile/types'
+import { useDispatchWeb } from 'app/hooks/useDispatchWeb'
+// import { usePushRouteWeb } from 'app/hooks/usePushRouteWeb'
+import { useSelectorWeb } from 'app/hooks/useSelectorWeb'
 import { useThemedStyles } from 'app/hooks/useThemedStyles'
-import { flexCol, flexRow } from 'app/styles'
+import { getPlaying, getPlayingUid } from 'app/store/audio/selectors'
 import { ThemeColors } from 'app/utils/theme'
 
 import { TrackBannerIcon, TrackBannerIconType } from './TrackBannerIcon'
 import { TrackTileBottomButtons } from './TrackTileBottomButtons'
 import { TrackTileCoSign } from './TrackTileCoSign'
+import { TrackTileContainer } from './TrackTileContainer'
 import { TrackTileMetadata } from './TrackTileMetadata'
 import { TrackTileStats } from './TrackTileStats'
 import { TrackTileTopRight } from './TrackTileTopRight'
 
 const createStyles = (themeColors: ThemeColors) =>
   StyleSheet.create({
-    container: {
-      ...flexRow(),
-      minHeight: 152,
-      borderColor: themeColors.neutralLight8,
-      backgroundColor: themeColors.white,
-      borderWidth: 1,
-      borderRadius: 8,
-      maxWidth: 400,
-      marginHorizontal: 'auto',
-      marginBottom: 12,
-      elevation: 3,
-      shadowOpacity: 0.15,
-      shadowOffset: { width: 0, height: 1 },
-      shadowRadius: 3
-    },
     mainContent: {
-      ...flexCol(),
       flex: 1
     }
   })
 
-export const TrackTile = ({
-  artistName,
-  coSign,
-  coverArtSizes,
-  duration,
-  fieldVisibility,
-  goToTrackPage,
-  goToArtistPage,
-  hasCurrentUserReposted,
-  hasCurrentUserSaved,
-  id,
+export const TrackTile = (props: TrackTileProps) => {
+  const { uid } = props
+
+  // Using isEqual as the equality function to prevent rerenders due to object references
+  // not being preserved when syncing redux state from client.
+  // This can be removed when no longer dependent on web client
+  const track = useSelectorWeb(state => getTrack(state, { uid }), isEqual)
+  const user = useSelectorWeb(state => getUserFromTrack(state, { uid }))
+  if (!track || !user) {
+    console.warn('Track or user missing for TrackTile, preventing render')
+    return null
+  }
+
+  return <TrackTileComponent {...props} track={track} user={user} />
+}
+
+const TrackTileComponent = ({
   index,
-  isArtistPick,
-  isOwner,
-  isPlaying,
   isTrending,
-  isUnlisted,
-  listenCount,
-  makeGoToFavoritesPage,
-  makeGoToRepostsPage,
-  onClickOverflow,
   onLoad,
-  onShare,
-  repostCount,
-  saveCount,
   showArtistPick,
   showRankIcon,
-  showSkeleton,
-  title,
   togglePlay,
-  toggleRepost,
-  toggleSave,
-  user,
-  uid
-}: TrackTileProps) => {
+  track,
+  uid,
+  user
+}: TrackTileProps & { track: Track; user: User }) => {
+  const {
+    // permalink,
```

Comment:
comment

Reply:
I left these in because they are needed for the routing todos, think it should be fine